File: day1/javascript/solution-part-two.js
const { readFileSync } = require("node:fs");
const input = readFileSync("../input.txt", "utf-8").split("\n");
let increased = 0,
  numbers = [],
  length = input.length;
for (let i = 0, preSum; i < length; i++) {
  if (!i) {
    numbers.push(+input[i], +input[i + 1], +input[i + 2]);
    preSum = numbers[0] + numbers[1] + numbers[2];
    continue;
  }
  numbers.push(+input[i + 2]);
  const curSum = numbers[i] + numbers[i + 1] + numbers[i + 2];
  preSum = curSum > preSum ? (increased++, curSum) : curSum;
}
console.log(increased);

File: day1/ruby/solution-part-two.rb
file = File.open('input.txt')
file_data = file.readlines.map(&:chomp)
increased = 0
preSum = nil
numbers = []
(0..file_data.length).each do |i|
  if i === 0
    numbers.push(file_data[i], file_data[i + 1], file_data[i + 2])
    preSum = numbers[0].to_i + numbers[1].to_i + numbers[2].to_i
    next
  end
  numbers.push(file_data[i + 2].to_i)
  curSum = numbers[i].to_i + numbers[i + 1].to_i + numbers[i + 2].to_i
  increased += 1 if curSum > preSum
  preSum = curSum
end
puts increased
